website/docs/icons/library/index.js

```diff
@@ -127,24 +128,36 @@ export default class Index extends Component {
 
   @action
   selectIconSize(event) {
+    const { value: selectedIconSize } = event.target;
+
     this.router.transitionTo({
       queryParams: {
+        selectedIconSize,
         searchQuery: this.searchQuery,
         selectedGroupType: this.selectedGroupType,
-        selectedIconSize: event.target.value,
       },
     });
+
+    this.eventTracking.trackEvent(
+      `Icon Library - Size Selector - ${selectedIconSize}`
+    );
   }
 
   @restartableTask *searchIcons(searchQuery) {
     yield timeout(DEBOUNCE_MS);
 
+    searchQuery = searchQuery !== '' ? searchQuery : null;
+
     this.router.transitionTo({
       queryParams: {
-        searchQuery: searchQuery !== '' ? searchQuery : null,
+        searchQuery,
         selectedGroupType: this.selectedGroupType,
         selectedIconSize: this.selectedIconSize,
       },
     });
+
+    if (searchQuery !== null) {
```

Comment:
@Dhaulagiri 

Do we want to track the search term being cleared? I assumed not, but can revert this if needed.

Reply:
I think that's fine not to log clearing the search. I do think we need to figure out a way to debounce or otherwise delay the event tracking more as 250ms is pretty easy to get into a situation where we are sending multiple events for the same search which is not really what we want.

Reply:
We could have a separately debounced function. Let me play around with it.

Reply:
👍  let me know what you find. I think ultimately I'd rather have no data than an overwhelming amount of partially/hopefully useful data.